springboot/src/main/java/ca/jrvs/apps/trading/dao/MarketDataDao.java

```diff
@@ -0,0 +1,209 @@
+     * @return a list of IEX quote object
+     * @throws IllegalArgumentException      if any ticker is invalid or ticker is empty
+     * @throws DataRetrievalFailureException if HTTP request is failed
+     */
+    @Override
+    public List<IexQuote> findAllById(Iterable<String> tickerList) throws IllegalArgumentException,
+            DataRetrievalFailureException {
+        int tickerNumber = 0;
+        for (String ticker : tickerList) {
+            if (!ticker.matches("[a-zA-Z]{2,4}")) {
+                throw new IllegalArgumentException("Illegal ticker format!");
+            }
+            tickerNumber++;
+        }
+        if (tickerNumber == 0) {
+            throw new IllegalArgumentException("No ticker number found");
+        }
+        List<IexQuote> quotes = new ArrayList<>();
+        String tickerListString = String.join(",", tickerList);
+        String url = String.format(IEX_BATCH_URL, tickerListString);
+        Optional<String> quoteString = null;
+        try {
+            quoteString = executeHttpGet(url);
+        } catch (Exception e) {
+            logger.error("Invalid url");
+            throw new DataRetrievalFailureException(e.getMessage());
+        }
+        JSONObject jsonObject = new JSONObject(quoteString.get());
+        ObjectMapper mapper = new ObjectMapper();
+        IexQuote quote;
+        for (String ticker : tickerList) {
+            String stringQuote = jsonObject.getJSONObject(ticker).getJSONObject("quote").toString();
+            try {
+                quote = mapper.readValue(stringQuote, IexQuote.class);
+            } catch (IOException ex) {
+                logger.error("Failed to convert JSON to quote object");
+                throw new JSONException(ex.getMessage());
+            }
+            quotes.add(quote);
+        }
+        return quotes;
+    }
+
+
+    /**
+     * Execute a get and return http entity/body as a string
+     * <p>
+     * Use EntityUtils.toString to process HTTP entity
+     *
+     * @param url resource URL
+     * @return HTTP response body or Optional.empty for 404 response
+     * @throws DataRetrievalFailureException if HTTP failed or status code is unexpected
+     */
+    private Optional<String> executeHttpGet(String url) throws DataRetrievalFailureException, MalformedURLException {
+        HttpClient httpClient = getHttpClient();
+        HttpGet httpRequest = new HttpGet(url);
+        HttpResponse httpResponse;
+        try {
+            httpResponse = httpClient.execute(httpRequest);
+        } catch (IOException e) {
+            logger.error("Invalid URL");
+            throw new MalformedURLException(e.getMessage());
+        }
+        int status = httpResponse.getStatusLine().getStatusCode();
+        if (status != HTTP_OK) {
+            throw new DataRetrievalFailureException("Unexpected http status" + status);
+        }
+        if (httpResponse.getEntity() == null) {
+            throw new RuntimeException("Empty response body");
```

Comment:
try not to throw RuntimeExcption, you need to be specific. 
Since you expect the http returns a body, so throws a `DataRetrievalFailureException` (as per Javadoc above)  if the HTTP body is missing.